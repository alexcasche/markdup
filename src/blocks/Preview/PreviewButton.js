import { Button } from 'elements'
import { tint, shade } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'

const MODIFIERS = {
  mobile: ({ theme }) => `
    color: pink;
    @media only screen and (min-width: ${theme.breakPoints.mobile}) {
      display: none;
    }
  `,
}

const PreviewButton = Button.extend`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  margin-left: 10px;
  background-color: ${props => tint(0.25, props.theme.colors.grey)};
  border: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  border-radius: 2px;
  z-index: 5;
  &:hover {
    background-color: ${props => tint(0.35, props.theme.colors.grey)};
  }
  svg {
    color: ${props => shade(0.9, props.theme.colors.grey)};
    height: 14px;
    width: auto !important;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default PreviewButton
