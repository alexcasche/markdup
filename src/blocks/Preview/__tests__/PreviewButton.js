import PreviewButton from '../PreviewButton'

describe('PreviewButton Element', () => {
  it('renders correctly', () => {
    const tree = render(<PreviewButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds mobile modifiers', () => {
    const tree = render(<PreviewButton modifiers={['mobile']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'none', {
      media: 'only screen and (min-width:600px)',
    })
  })
})
