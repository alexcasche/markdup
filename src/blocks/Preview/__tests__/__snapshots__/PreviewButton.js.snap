// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewButton Element adds mobile modifiers 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  margin-left: 10px;
  background-color: #eaeaec;
  border: 1px solid #d5d6d9;
  border-radius: 2px;
  z-index: 5;
  color: pink;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover {
  background-color: #e1e2e4;
}

.c0 svg {
  color: #999ba1;
  height: 14px;
  width: auto !important;
}

@media only screen and (min-width:600px) {
  .c0 {
    display: none;
  }
}

<button
  className="c0"
/>
`;

exports[`PreviewButton Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  font-family: "IBM Plex Sans Condensed",Helvetica,Arial,sans-serif;
  font-weight: 500;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  margin-left: 10px;
  background-color: #eaeaec;
  border: 1px solid #d5d6d9;
  border-radius: 2px;
  z-index: 5;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover {
  background-color: #e1e2e4;
}

.c0 svg {
  color: #999ba1;
  height: 14px;
  width: auto !important;
}

<button
  className="c0"
/>
`;
